api/coreservice.go

```diff
@@ -1590,9 +1590,31 @@ func (core *coreService) BlockChain() blockchain.Blockchain {
 	return core.bc
 }
 
-// StateFactory return the member sf
-func (core *coreService) StateFactory() factory.Factory {
-	return core.sf
+func (core *coreService) SimulateExecution(ctx context.Context, addr address.Address, exec *iotextypes.Execution) ([]byte, *action.Receipt, error) {
+	amount, ok := new(big.Int).SetString(exec.GetAmount(), 10)
+	if !ok {
+		return nil, nil, errors.New("failed to set execution amount")
+	}
+	state, err := accountutil.AccountState(core.sf, addr)
+	if err != nil {
+		return nil, nil, err
+	}
+	ctx, err = core.bc.Context(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	sc, err := action.NewExecution(
+		exec.GetContract(),
+		state.Nonce+1,
+		amount,
+		core.cfg.Genesis.BlockGasLimit,
```

Comment:
Current func used for `TraceTransactionStructLogs` is ok. 
But let's make this func usable in more scenario: an error should be returned if the the gaslimit or gasprice of exec is inadaquate. it is because we can't assume that the gaslimit and gasprice of input are reasonable.
See 
https://github.com/iotexproject/iotex-core/pull/3049/files#diff-a84e0d7fdb009bebf88eb3c2582dc4c2f5fce09f6d40a3a514b553c198675f42R383
https://github.com/iotexproject/iotex-core/pull/3049/files#diff-6dc4e8a6418cb4f7e5b7da626b745fb04ff21d7c37b77d883b8c8b9ab7aacfcbR985
https://github.com/iotexproject/iotex-core/pull/3049/files#diff-6dc4e8a6418cb4f7e5b7da626b745fb04ff21d7c37b77d883b8c8b9ab7aacfcbR1006

Reply:
Not the duty of this PR. I will leave this to you.